src/browser/service/fetch.rs

```diff
@@ -550,42 +549,57 @@ impl Request {
         let fetch_result = fetch_object.result;
         let request = fetch_object.request;
 
+        // FIXME What is this giant gross match ladder?
```

Comment:
See the top of the file - the module is deprecated. 

There was a bunch of fetch API versions, some of them were limited by missing async support in Rust or had other problems that resulted in such monsters.

I think you can remove the file and add probably a breaking change into the changelog.